src/compiler/checker.ts

```diff
@@ -24365,9 +24365,24 @@ namespace ts {
                 callIsIncomplete = node.arguments.end === node.end;
 
                 // If a spread argument is present, check that it corresponds to a rest parameter or at least that it's in the valid range.
-                const spreadArgIndex = getSpreadArgumentIndex(args);
-                if (spreadArgIndex >= 0) {
-                    return spreadArgIndex >= getMinArgumentCount(signature) && (hasEffectiveRestParameter(signature) || spreadArgIndex < getParameterCount(signature));
+                const firstSpreadArgIndex = getSpreadArgumentIndex(args);
+                const existSpreadArgIndex = firstSpreadArgIndex >= 0;
+                if (existSpreadArgIndex) {
+                      if (firstSpreadArgIndex === args.length - 1) {
+                          return firstSpreadArgIndex >= getMinArgumentCount(signature) && (hasEffectiveRestParameter(signature) || firstSpreadArgIndex < getParameterCount(signature));
+                      }
+
+                      const totalCount = node.arguments.reduce((total, next) => {

```

Comment:
About using `reduce`: first, there is a preference for plain loops in the code, and there's another minor issue: you're using `getSpreadArgumentIndex` which will scan the list and test members with `isSpreadArgument`, and then the `reduce` tests again for `isSpreadArgument`.  If you switch to a plain loop, you can skip the re-test for all of these elements.

Reply:
I changed reduce to a plain loop. And I tried to remove Spread Argument but it failed. I think getSpreadArgumentIndex only searches the first spread element. So it needs to use isSpreadArgument for the case using two or more spread elements.